Libplanet/Net/Swarm.cs

```diff
@@ -2053,60 +2053,78 @@ private void DoReply(object sender, NetMQQueueEventArgs<Message> e)
             while (!cancellationToken.IsCancellationRequested)
             {
                 _logger.Verbose("Waiting for new request...");
-                var req = await _requests.TakeAsync(cancellationToken);
-                _logger.Verbose("Request {RequestId} taken.", req.Id);
+                MessageRequest req = await _requests.TakeAsync(cancellationToken);
 
-                using (var dealer = new DealerSocket(ToNetMQAddress(req.Peer)))
+                try
+                {
+                    await ProcessRequest(req, cancellationToken);
+                }
+                catch (Exception e)
                 {
-                    dealer.Options.Linger = Timeout.InfiniteTimeSpan;
                     _logger.Debug(
-                        "Trying to send {Message} to {PeerAddress}...",
-                        req.Message,
-                        req.Peer.Address
+                        "Unexpected exception occurred during {ProcessRuntime}(). {e}",
+                        nameof(ProcessRuntime),
+                        e);
+                    await _requests.AddAsync(req, cancellationToken);
+                    await Task.Delay(100, cancellationToken);
```

Comment:
How about using `final` instead of delay in both `ProcessRequest()` and `catch` block?

Reply:
`Task.Delay` in `ProcessRequest()` is for delaying dealer disposing to avoid `ObjectDisposedException` on `NetMQPoller`. So they have different purpose.